app/src/org/commcare/connect/network/base/BaseApiCallback.kt

```diff
@@ -67,6 +68,15 @@ abstract class BaseApiCallback<T>(val baseApiHandler: BaseApiHandler<T>) :
                         null
                     )
                     true
+                } else if (errorCode.equals("INTEGRITY_ERROR", ignoreCase = true)) {
+                    if (json.has("sub_code")) {
+                        Logger.log(LogTypes.TYPE_EXCEPTION, json.optString("sub_code"))
```

Comment:
not enough details in the log, change to - 
` Logger.log(LogTypes.MAINTENANCE, "Integrity error with subcode " + json.optString("sub_code"))`